app/Http/Controllers/ConversationController.php

```diff
@@ -23,7 +23,7 @@ public function index()
 
     public function store()
     {
-        $participants = [auth()->user()->id];
+        $participants = [auth()->user()];
```

Comment:
We also need the user that it wants to have a conversation with, right?

Reply:
So this is just for demo purpose. Just like when you start a slack channel you get added to the conversation and then you can use the `join` function to add additional users

Reply:
Ahh, make sense :)

Reply:
The way im doing it ATM. is to check if the chat is started from the client table or the user table, like this

```php
    public function store()
    {
        $participant_id = request()->participant_id;
        $participant_class = request()->participant_class;

        if ($participant_class === 'client') {
            $participant = Client::find($participant_id);
        } else {
            $participant = User::find($participant_id);
        }

        $participants = [auth()->user(), $participant];
        $conversation = Chat::createConversation($participants);

        return response($conversation);
    }
```

Then the frontend send a request with the id and the class name

Reply:
Yes that makes sense. If you have a bunch of participant types you may want to pass the full a fully qualified namespace for `participant_class`

```php
    public function store()
    {
        $participantId = request()->participant_id;
        $participantClass = request()->participant_class;
        
        $participants = [auth()->user()]

        if ($participantClass) {
           // An exception will be thrown for invalid classes or you can add validation for allowed classes
           $participants[] = app($participantClass)->find($participantId);
        }

        $conversation = Chat::createConversation($participants);

        return response($conversation);
    }
```